packages/realm/src/Configuration.ts

```diff
@@ -197,155 +193,73 @@ type BaseConfiguration = {
 //   disableFormatUpgrade?: boolean;
 // };
 
-// Need to use `CanonicalObjectSchema` rather than `ObjectSchema` due to some
-// integration tests using `openRealmHook()`. That function sets `this.realm`
-// to the opened realm whose schema is a `CanonicalObjectSchema[]`. Consequently,
-// the key `"ctor"` (which doesn't exist on `ObjectSchema`) also needs to be allowed.
-const OBJECT_SCHEMA_KEYS = new Set<keyof CanonicalObjectSchema>([
-  "name",
-  "primaryKey",
-  "embedded",
-  "asymmetric",
-  "properties",
-  // Not part of `ObjectSchema`
-  "ctor",
-]);
-
-// Need to use `CanonicalObjectSchemaProperty` rather than `ObjectSchemaProperty`
-// due to the same reasons as above.
-const PROPERTY_SCHEMA_KEYS = new Set<keyof CanonicalObjectSchemaProperty>([
-  "type",
-  "objectType",
-  "property",
-  "default",
-  "optional",
-  "indexed",
-  "mapTo",
-  // Not part of `ObjectSchemaProperty`
-  "name",
-]);
-
 /**
  * Validate the fields of a user-provided Realm configuration.
  */
 export function validateConfiguration(config: unknown): asserts config is Configuration {
-  assert.object(config);
-  const { path, schema, onMigration, sync } = config;
-  if (typeof onMigration !== "undefined") {
-    assert.function(onMigration, "migration");
+  assert.object(config, "realm configuration", { allowArrays: false });
```

Comment:
Would like to in the future change the `assert.object()` to default `allowArrays` to `true`.